Makefile

```diff
@@ -1,96 +1,106 @@
-
-GCR_URL = us.gcr.io/vcm-ml
-EXPERIMENT ?=c12_6ranks_standard
 SHELL=/bin/bash
 CWD=$(shell pwd)
+CMD ?= bash
 DEV ?=y
+ROOT_DIR ?= /pyFV3
+
 NUM_RANKS ?=6
-VOLUMES ?=
-CONTAINER_ENGINE ?=docker
+MPIRUN_ARGS ?=--oversubscribe --mca btl_vader_single_copy_mechanism none
+MPIRUN_CALL ?=mpirun -np $(NUM_RANKS) $(MPIRUN_ARGS)
+
+TEST_DATA_LOC ?=test_data/
+TEST_DATA_VERSION ?=8.1.3
+TEST_DATA_HOST ?= https://portal.nccs.nasa.gov/datashare/astg/smt/pace-regression-data/
+TEST_TYPE ?= standard
+TEST_RESOLUTION ?= c12
+TEST_CONFIG ?= $(TEST_RESOLUTION)_$(NUM_RANKS)ranks
+TEST_CASE ?=$(TEST_CONFIG)_$(TEST_TYPE)
+TEST_ARGS ?=-vs
+PORT ?=8888
 RUN_FLAGS ?=--rm
 RUN_FLAGS += -e FV3_DACEMODE=$(FV3_DACEMODE)
-TEST_ARGS ?=
-TARGET ?=dycore
-FV3UTIL_DIR=$(CWD)/external/util
-ROOT_DIR:=$(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
-TEST_DATA_ROOT ?=$(ROOT_DIR)/test_data/
-TEST_DATA_FTP ?=in/put/abc/cosmo/fuo/pace/fv3core/
-
-FV3=fv3core
-FV3_PATH ?=/pace/$(FV3)
-
-ifeq ($(DEV),y)
-	VOLUMES += -v $(CWD)/..:/pace
-endif
-CONTAINER_CMD?=$(CONTAINER_ENGINE) run $(RUN_FLAGS) $(VOLUMES) $(PACE_IMAGE)
+IMAGE_NAME ?= noaa-gfdl/pyfv3
+APP_NAME ?=pyFV3_dev
+SAVEPOINT_SETUP=pip list
+THRESH_ARGS=--threshold_overrides_file=$(ROOT_DIR)/tests/savepoint/translate/overrides/$(TEST_TYPE).yaml
 
-ifneq (,$(findstring $(PACE_IMAGE),$(CONTAINER_CMD)))
-	TEST_DATA_RUN_LOC =/test_data
-else
-	TEST_DATA_RUN_LOC=$(TEST_DATA_HOST)
-endif
+BUILD_FLAGS ?=
+VOLUMES ?=
 
-PYTHON_FILES = $(shell git ls-files | grep -e 'py$$' | grep -v -e '__init__.py')
-PYTHON_INIT_FILES = $(shell git ls-files | grep '__init__.py')
-TEST_DATA_TARFILE=dat_files.tar.gz
-TEST_DATA_TARPATH=$(TEST_DATA_HOST)/$(TEST_DATA_TARFILE)
-MPIRUN_ARGS ?=--oversubscribe
-MPIRUN_CALL ?=mpirun -np $(NUM_RANKS) $(MPIRUN_ARGS)
-BASE_INSTALL?=$(FV3)-install-serialbox
-DATA_BUCKET= $(REGRESSION_DATA_STORAGE_BUCKET)/$(FORTRAN_SERIALIZED_DATA_VERSION)/$(EXPERIMENT)/$(TARGET)/
-
-TEST_TYPE=$(word 3, $(subst _, ,$(EXPERIMENT)))
-THRESH_ARGS=--threshold_overrides_file=$(FV3_PATH)/tests/savepoint/translate/overrides/$(TEST_TYPE).yaml
-PYTEST_MAIN=pytest $(TEST_ARGS) $(FV3_PATH)/tests/main
-PYTEST_SEQUENTIAL=pytest --data_path=$(TEST_DATA_RUN_LOC) $(TEST_ARGS) $(THRESH_ARGS) $(FV3_PATH)/tests/savepoint
-# we can't rule out a deadlock if one test fails, so we must set maxfail=1 for parallel tests
-PYTEST_PARALLEL=$(MPIRUN_CALL) python3 -m mpi4py -m pytest --maxfail=1 --data_path=$(TEST_DATA_RUN_LOC) $(TEST_ARGS) $(THRESH_ARGS) -m parallel $(FV3_PATH)/tests/savepoint
-ifeq ($(DEV),y)
-	VOLUMES += -v $(CWD)/$(FV3):/$(FV3)/$(FV3) -v $(CWD)/tests:/$(FV3)/tests -v $(FV3UTIL_DIR):/external/util -v $(CWD)/external/dsl:/external/dsl -v $(CWD)/external/stencils:/external/stencils
+ifeq ($(DEV), y)
+	VOLUMES += -v $(CWD):/pyFV3
 endif
-CONTAINER_CMD?=$(CONTAINER_ENGINE) run $(RUN_FLAGS) $(VOLUMES) $(PACE_IMAGE)
 
-clean:
+TEST_DATA_TARFILE = $(TEST_DATA_VERSION)_$(TEST_RESOLUTION)_$(NUM_RANKS)_ranks_$(TEST_TYPE).tar.gz
```

Comment:
Yeah that's nasty, my bad. I should clean naming before we merge.

Reply:
thank you 🙏

Reply:
Done